### PR TITLE
net: context: connect: Make sure local end is bound before connecting

### PR DESCRIPTION
**Pull request by pfalcon**
Introduce net_context_bind_default() to ensure that local address is
set for context if not yet (via explict bind() call). This fixes
dereferences of NULL pointer to local address which was exposed when
MMU was enabled for qemu_x86.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

### CODE REVIEW

**Comment by pfalcon** [comment]
This version supersedes https://github.com/zephyrproject-rtos/zephyr/pull/785

File: subsys/net/ip/net_context.c [hunk 1]
```diff
@@ -689,6 +689,53 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+/* If local address is not bound, bind it to INADDR_ANY and random port. */
+static int bind_default(struct net_context *context)
+{
+	sa_family_t af = net_context_get_family(context);
```
Comment (jukkar) [inline subsys/net/ip/net_context.c]: Would prefer "family" instead of "af", as the family variable is used in other part of the code.

**Review by jukkar — CHANGES_REQUESTED** [review]
LGTM, very minor nit pick

File: subsys/net/ip/net_context.c (hunk 1 above)
Comment (pfalcon) [inline subsys/net/ip/net_context.c]: Fixed

**Review by jukkar — APPROVED** [review]

**Pull request merged**